tasks/Resources/Provisioning/Hive/provision.robot

```diff
@@ -0,0 +1,91 @@
+    Oc Apply    kind=ClusterClaim    src=tasks/Resources/Provisioning/Hive/claim.yaml
+    ...    template_data=${infrastructure_configurations}
+    Wait Until Keyword Succeeds    30 min    10 s
+    ...    Confirm Cluster Is Claimed
+
+Does ClusterName Exists
+    @{clusterpoolname} =    Oc Get   kind=ClusterPool    namespace=rhods    api_version=hive.openshift.io/v1
+    Log Many    @{clusterpoolname}
+    Log Many    ${infrastructure_configurations['hive_cluster_name']}
+    FOR    ${name}    IN    @{clusterpoolname}
+        IF    "${name}[metadata][name]" == "${infrastructure_configurations['hive_cluster_name']}"
+            Log    ${name}[metadata][name]    console=True
+            ${clustername_exists} =    Set Variable    "${name}[metadata][name]"
+            Return From Keyword    ${clustername_exists}
+        END  
+    END
+    [Return]    False
+
+Get Clusters
+    @{clusters} =    Oc Get    kind=ClusterClaim    namespace=rhods
+    FOR   ${cluster}    IN    @{clusters}
+        Log    Name: "${cluster}[spec][clusterPoolName]"    console=True
+    END
+
+Get Cluster Credentials
+    Set Log Level    None
+    ${ns} =    Oc Get    kind=Namespace    label_selector=hive.openshift.io/cluster-pool-name=${infrastructure_configurations}[hive_cluster_name]
+    ${consoleURL} =    Run and Return Rc And Output    oc -n ${ns[0]['metadata']['name']} get cd ${ns[0]['metadata']['name']} -o jsonpath='{ .status.webConsoleURL }'
+    Create File  cluster_details.txt  consoleUrl=${consoleURL}\n
+    ${credentials} =    Run and Return Rc And Output    oc extract -n ${ns[0]['metadata']['name']} secret/$(oc -n ${ns[0]['metadata']['name']} get cd ${ns[0]['metadata']['name']} -o jsonpath='{.spec.clusterMetadata.adminPasswordSecretRef.name}') --to=-
+    ${credentials_splited} =    Split To Lines    ${credentials[1]}
+    Append to File  cluster_details.txt  username=${credentials_splited[3]}\n
+    Append to File  cluster_details.txt  password=${credentials_splited[1]}\n
+
+Login To Cluster
+    Set Log Level    None
+    ${ns} =    Oc Get    kind=Namespace    label_selector=hive.openshift.io/cluster-pool-name=${infrastructure_configurations['hive_cluster_name']}
+    ${ClusterDeployment} =    Oc Get    kind=ClusterDeployment    name=${ns[0]['metadata']['name']}    
+    ...    namespace=${ns[0]['metadata']['name']}    api_version=hive.openshift.io/v1
+    ${apiURL} =    Set Variable    "${ClusterDeployment[0]['status']['apiURL']}"
+    ${credentials} =    Run and Return Rc And Output    oc extract -n ${ns[0]['metadata']['name']} secret/$(oc -n ${ns[0]['metadata']['name']} get cd ${ns[0]['metadata']['name']} -o jsonpath='{.spec.clusterMetadata.adminPasswordSecretRef.name}') --to=-
+    ${credentials_splited} =    Split To Lines    ${credentials[1]}
+    Run And Return Rc    oc login --username=${credentials_splited[3]} --password=${credentials_splited[1]} ${apiURL} --insecure-skip-tls-verify
+    Log    Logged in to Hive cluster    console=True
```

Comment:
is it loggin into the created gcp cluster or into the cluster which creates the gcp cluster?

Reply:
into the created GCP cluster